blackmarket/UI/Common/StateButton.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  StateButton.swift
+//  blackmarket
+//
+//  Created by Daniel Parra on 9/28/22.
+//
+
+import SwiftUI
+
+struct StateButton: View {
+    let action: () -> ()
+    let title: String
+    var isEnabled: Bool
+
+    var body: some View {
+            Button(action: {
+                withAnimation {
+                    action()
+                }
+            }) {
+                Text(title)
+                    .font(.subheadline)
+                    .frame(maxWidth: .infinity, maxHeight: 45)
+                    .fontWeight(.bold)
+                    .foregroundColor(
+                        isEnabled ?
+                        .init(hex: "ffffff") :
+                        .init(hex: "636363")
```

Comment:
```suggestion
                        isEnabled ? .white : .disabledGray
```

end move these to the color constants